app/src/components/datasets/DatasetContentTabs/General/DatasetEntriesTable/parseableToToolCalls.ts

```diff
@@ -0,0 +1,70 @@
+// Check if the str could be parsed to an array of tool calls
+export const parseableToToolCalls = (str: string) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let parsedJSON: any;
+  try {
+    parsedJSON = JSON.parse(str);
+  } catch {
+    return false;
+  }
+
+  // Check if the parsedJSON is an object and not null
+  if (parsedJSON === null || !Array.isArray(parsedJSON)) {
```

Comment:
I think this function would be a lot shorter, cleaner and more maintainable if it were just two zod validators, one for the array of objects, and then a second one for the arguments?

Reply:
Ah yes, I had forgotten how wonderful zod is.